hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/SparkBaseIndexSupport.scala

```diff
@@ -77,20 +77,24 @@ abstract class SparkBaseIndexSupport(spark: SparkSession,
 
   def invalidateCaches(): Unit
 
-  protected def getPrunedFileNames(prunedPartitionsAndFileSlices: Seq[(Option[BaseHoodieTableFileIndex.PartitionPath], Seq[FileSlice])],
-                                   includeLogFiles: Boolean = false): Set[String] = {
-    prunedPartitionsAndFileSlices
+  protected def getPrunedPartitionsAndFileNames(prunedPartitionsAndFileSlices: Seq[(Option[BaseHoodieTableFileIndex.PartitionPath], Seq[FileSlice])],
+                                                includeLogFiles: Boolean = false): (Set[String], Set[String]) = {
+    val prunedPartitions : scala.collection.mutable.Set[String] = scala.collection.mutable.Set.empty[String]
+    val prunedFiles = prunedPartitionsAndFileSlices
       .flatMap {
         case (_, fileSlices) => fileSlices
       }
       .flatMap { fileSlice =>
+        prunedPartitions.add(fileSlice.getPartitionPath)
```

Comment:
aren't we already getting partition path in the left side of each tuple in `(Option[BaseHoodieTableFileIndex.PartitionPath], Seq[FileSlice])`? If we can use that then we don't need to allocate and keep adding to prunedPartitions.